src/lib/stitching/exchange/__tests__/stitching.test.ts

```diff
@@ -0,0 +1,121 @@
+import { getFieldsForTypeFromSchema } from "lib/stitching/lib/getTypesFromSchema"
+import {
+  getExchangeMergedSchema,
+  getExchangeStitchedSchema,
+} from "./testingUtils"
+
+it("extends the Order objects", async () => {
+  const mergedSchema = await getExchangeMergedSchema()
+
+  const orderables = ["CommerceBuyOrder", "CommerceOfferOrder", "CommerceOrder"]
+  for (const orderable of orderables) {
+    const orderableFields = await getFieldsForTypeFromSchema(
+      orderable,
+      mergedSchema
+    )
+
+    expect(orderableFields).toContain("buyerDetails")
+    expect(orderableFields).toContain("sellerDetails")
+
+    // Any field inside the CommerceBuyOrder & CommerceOfferOrder which
+    // ends in cents should have a version without cents which is a
+    // string equivalent
+    const fieldsWithCents = orderableFields.filter(f => f.endsWith("Cents"))
+    for (const field of fieldsWithCents) {
+      expect(orderableFields).toContain(field.replace("Cents", ""))
```

Comment:
It worries me that we're assuming whole dollar values anywhere. (Am I even understanding that correctly?)

Reply:
I think that ship has sailed, this is aiming to replicate the setup we have everywhere in MP ( which is using `xxYyCents` and then exposing `xxYy` as a string version where you have to pass in a currency signifier ) 

That said, this test introspects the classes from exchange to verify we do that for everything that has a cents (which happily caught one we missed) which I think is the right kind of test to write (as it means someone can add a new currency field and the stitching tests will fail)

Reply:
Are the `xxYy` strings meant to reflect the "formatted" values, or are they expected to be whole numbers only?

I'd love to identify any locations that would actually fail or be wrong when there are non-whole price values, since I'm sure that will be a painful bridge to cross later.

Reply:
Those strings are the formatted values